python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py

```diff
@@ -1,19 +1,39 @@
-from typing import Optional
+from collections import namedtuple
+from datetime import datetime
+from typing import List, Optional
 
 import boto3
 import botocore
 import dagster._check as check
+from dagster._annotations import deprecated
 
+Object = namedtuple("Object", ["key", "last_modified"])
 
-def get_s3_keys(
+
+def get_objects(
     bucket: str,
     prefix: str = "",
     since_key: Optional[str] = None,
+    since_last_modified: Optional[str] = None,
     s3_session: Optional[boto3.Session] = None,
```

Comment:
I suggest renaming this to `client` and dropping the type annotation since boto3's clients are really untyped. 

```python
def get_objects(
...,
client: Any = None
):
    client = client or boto3.client("s3")
    ...

Reply:
One of my other PRs is about to add `boto3-stubs-lite[s3]` to `dagster-aws` tho. 
After this is done, you could do the following: 
```python
if TYPE_CHECKING
    from mypy_boto3_s3.client import S3Client
    
def get_objects(
    ...,
    client: Optional["S3Client"] = None
    ): ...
```